DependencyInjection/CmfSeoExtension.php

```diff
@@ -290,4 +296,29 @@ private function loadSitemapHandling($config, XmlFileLoader $loader, ContainerBu
             $container->removeDefinition($this->getAlias().'.sitemap.guesser.alternate_locales');
         }
     }
+
+    /**
+     * Configuration block to configure form.
+     *
+     * @param array            $config
+     * @param ContainerBuilder $container
+     */
+    private function loadFormConfiguration($config, ContainerBuilder $container)
+    {
+        $seoMetadataClass = null;
+        if (isset($config['data_class'])
+            && isset($config['data_class']['seo_metadata'])
+            && null !== $config['data_class']['seo_metadata']
+        ) {
+            $seoMetadataClass = $config['data_class']['seo_metadata'];
+        } else {
+            if ($container->hasParameter($this->getAlias() . '.backend_type_phpcr')) {
```

Comment:
this is a bit weird. i would have a parameter `$storage` to this method which you pass as `$this->isConfigEnabled($container, $config['persistence']['phpcr'] ? 'phpcr' : 'default'`. then you can determine the class based on the explicit argument.